File: app/src/main/java/com/example/android_challenge_btg/Activities/MainActivity.java
package com.example.android_challenge_btg.Activities;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.example.android_challenge_btg.R;
import com.example.android_challenge_btg.ViewPagerAdapter;
import com.example.android_challenge_btg.fragments.MoviesFragment;
import com.google.android.material.tabs.TabLayout;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tabCategories)
    TabLayout tabLayout;
    @BindView(R.id.viewPager)
    ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setupViewPager();
    }

    private void setupViewPager(){
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        MoviesFragment popularMoviesFragment = new MoviesFragment();
        MoviesFragment favoritesFragment = new MoviesFragment();
        popularMoviesFragment.setFavorites(false);
        favoritesFragment.setFavorites(true);
        adapter.addFragment(popularMoviesFragment, "Movies");
        adapter.addFragment(favoritesFragment, "Favorites");
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: app/src/main/java/com/brunoalmeida/movies/ui/main/MainActivity.kt
package com.brunoalmeida.movies.ui.main

import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import androidx.room.Room
import com.brunoalmeida.movies.R
import com.brunoalmeida.movies.data.AppDatabase
import com.brunoalmeida.movies.presentation.genders.GendersViewModel
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: GendersViewModel

    companion object {
        var database: AppDatabase? = null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        database = Room.databaseBuilder(this, AppDatabase::class.java, "tmdb").allowMainThreadQueries().fallbackToDestructiveMigration().build()

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val sectionsPagerAdapter = SectionsPagerAdapter(this, supportFragmentManager)
        val viewPager: ViewPager = findViewById(R.id.view_pager)
        viewPager.adapter = sectionsPagerAdapter
        val tabs: TabLayout = findViewById(R.id.tabs)
        tabs.setupWithViewPager(viewPager)
        setSupportActionBar(toolbar)

        viewModel = ViewModelProviders.of(this).get(GendersViewModel::class.java)
        viewModel.getGenders()


    }


}